src/layers/conv.jl

```diff
@@ -1452,3 +1452,200 @@ function Base.show(io::IO, l::EGNNConv)
     end
     print(io, ")")
 end
+
+
+@doc raw"""
+    TransformerConv((in, ein) => out; heads=1, concat=true, init=glorot_uniform,
+        add_self_loops=false, bias_qkv=false, bias_root=true, root_weight=true, beta=false)
+
+The transformer-like multi head attention convolutional operator from the 
+[Masked Label Prediction: Unified Message Passing Model for Semi-Supervised 
+Classification](https://arxiv.org/abs/2009.03509) paper, which also considers 
+edge features.
+It further contains options to also be configured as the transformer-like convolutional operator from the 
+[Attention, Learn to Solve Routing Problems!](https://arxiv.org/abs/1706.03762) paper,
+including a successive feed-forward network as well as skip layers and batch normalization.
+
+The layer's basic forward pass is given by
+```math
+x_i' = W_1x_i + \sum_{j\in N(i)} \alpha_{ij} (W_2 x_j + W_6e_{ij})
+```
+where the attention scores are
+```math
+\alpha_{ij} = \mathrm{softmax}\left(\frac{(W_3x_i)^T(W_4x_j+
+W_6e_{ij})}{\sqrt{d}}\right).
+```
+
+Optionally, a combination of the aggregated value with transformed root node features 
+by a gating mechanism via
+```math
+x'_i = \beta_i W_1 x_i + (1 - \beta_i) \underbrace{\left(\sum_{j \in \mathcal{N}(i)}
+\alpha_{i,j} W_2 x_j \right)}_{=m_i}
+```
+with
+```math
+\beta_i = \textrm{sigmoid}(W_5^{\top} [ W_1 x_i, m_i, W_1 x_i - m_i ]).
+```
+can be performed.
+
+# Arguments 
+
+- `in`: Dimension of input features, which also corresponds to the dimension of 
+    the output features.
+- `ein`: Dimension of the edge features; if 0, no edge features will be used.
+- `heads`: Number of heads in output.
+- `concat`: Concatenate layer output or not. If not, layer output is averaged
+    over the heads.
+- `init`: Weight matrices' initializing function.
+- `add_self_loops`: Add self loops to the input graph.
+- `bias_qkv`: If set, bias is used in the key, query and value transformations for nodes.  
+- `bias_root`: If set, the layer will also learn an additive bias for the root when root 
+    weight is used.
+- `root_weight`: If set, the layer will add the transformed root node features
+    to the output.
+- `beta`: If set, will combine aggregation and transformed root node features by
+    gating mechanism.
```

Comment:
better call this option `gating` instead of `beta`.

Reply:
Yes, `gating is better!